### PR TITLE
feat: add cold-start step profile mode

### LINKED ISSUE #3123: performance: add explicit cold-start step profile mode
## Goal / Problem

Add an explicit cold-start or first-step profile mode for the performance smoke test.

## Context

The #3025 step-profile warmup slice makes `--step-profile` default to one untimed warmup step so large-crowd profiles expose steady-step hotspots instead of first-step compilation/setup overhead. That is useful for #3025 steady-state optimization work, but it means cold-start/JIT attribution should have a named mode rather than being inferred from `--warmup-steps 0`.

## Scope

- Add a clear CLI option or mode name for cold-start step profiling.
- Preserve the existing JSON schema fields (`measurement_mode`, `warmup_excluded`, warmup timing fields) or extend them compatibly if needed.
- Add focused tests proving the cold-start profile includes the first measured step and does not auto-inject a warmup.
- Document the intended use: cold-start/JIT/setup attribution, not steady-step bottleneck optimization.

## Acceptance Criteria

- A user can run the performance smoke test and request first-step/cold-start hotspots explicitly.
- The default steady-step profile behavior remains clear and tested.
- The output labels make it impossible to confuse cold-start and steady-step timings.
- Focused tests and ruff checks pass.

## Validation

- `uv run pytest tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py -q`
- `uv run ruff check scripts/validation/performance_smoke_test.py tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py`
- `uv run ruff format --check scripts/validation/performance_smoke_test.py tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py`

### PR DESCRIPTION
**Pull request by ll7**
## Summary

- add `--step-profile-mode {steady,cold-start}` to make cold-start step profiling explicit
- preserve the default steady profile behavior: omitted warmup plus `--step-profile` still uses one warmup step
- add focused tests for steady mode, cold-start mode, explicit warmup preservation, explicit zero warmup, and unknown-mode rejection

## Issue

Closes #3123. This is diagnostic tooling only; it does not claim a simulator speedup.

## Evidence

- `--step-profile-mode cold-start` now routes with zero warmup when `--warmup-steps` is omitted.
- `/tmp/3123_cold_start_profile.json` showed `measurement_mode=cold_only`, `warmup_excluded=false`, `first_step_sec=5.518s`, `step_loop_sec=5.529s`, `steady_step_loop_sec=0.011s`, scenario `dense_pedestrian_stress`, `pedestrian_count=17`, and top hotspot `robot_sf/gym_env/robot_env.py:819 step`.

## Validation

- `scripts/dev/run_worktree_shared_venv.sh -- uv run pytest tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py -q` (16 passed)
- `scripts/dev/run_worktree_shared_venv.sh -- uv run ruff check scripts/validation/performance_smoke_test.py tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py`
- `scripts/dev/run_worktree_shared_venv.sh -- uv run ruff format --check scripts/validation/performance_smoke_test.py tests/validation/test_performance_smoke_test.py tests/perf/test_large_crowd_step_profile_contract.py`
- `scripts/dev/run_worktree_shared_venv.sh -- uv run python scripts/validation/performance_smoke_test.py --large-crowd-profile --step-profile --step-profile-mode cold-start --json-output /tmp/3123_cold_start_profile.json`
- `git diff --check`

## Downstream Propagation

- #3123 should close on merge.
- Parent #3025 remains open because this adds the missing cold-start diagnostic mode; runtime optimization remains separate work.

## Research Result Guidance

- target claim/hypothesis/blocker: distinguish cold-start/JIT profile attribution from steady-step profile attribution for #3025.
- comparator/baseline: previous workflow required knowing to pass `--warmup-steps 0`; now a named mode expresses the intent.
- evidence tier: diagnostic tooling evidence only.
- result classification: support/tooling completion for #3123; no benchmark-strength speedup claim.
- decision/stop rule: cold-start mode emits `cold_only`/`warmup_excluded=false` and captures first-step hotspots.
- synthesis surface/update target: #3123 closed; #3025 remains the parent optimization surface.

### CODE REVIEW

File: scripts/validation/performance_smoke_test.py [hunk 1]
```diff
@@ -851,6 +871,25 @@ def parse_args() -> argparse.Namespace:
...
+
+    if step_profile_mode not in _STEP_PROFILE_MODES:
+        raise ValueError("step_profile_mode must be one of: steady, cold-start")
+    if warmup_steps is not None:
+        return warmup_steps
+    if not step_profile:
+        return 0
+    if step_profile_mode == _STEP_PROFILE_MODE_COLD_START:
+        return 0
+    return 1
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/validation/performance_smoke_test.py:893]: Addressed in a3f8492: _resolve_step_profile_warmup_steps now returns explicit warmup_steps before mode validation, defaults None to steady, and has regression coverage for explicit warmup with an otherwise unknown mode.
